[MRG][ENH]: Add Ability to export STC files as GIFTI (#12309)
mne-tools/mne-python: doc/changes/devel.rst

- Add ``method="polyphase"`` to :meth:`mne.io.Raw.resample` and related functions to allow resampling using :func:`scipy.signal.upfirdn` (:gh:`12268` by `Eric Larson`_)
- The package build backend was switched from ``setuptools`` to ``hatchling``. This will only affect users who build and install MNE-Python from source. (:gh:`12269`, :gh:`12281` by `Richard Höchenberger`_)
- :meth:`mne.Annotations.to_data_frame` can now output different formats for the ``onset`` column: seconds, milliseconds, datetime objects, and timedelta objects. (:gh:`12289` by `Daniel McCloy`_)

> - Add ``save_as_surface`` method to allow saving GIFTI files from STC SourceEstimates in :class:`~mne._BaseSurfaceSourceEstimate` (:gh:`12309` by `Peter Molfese`_)

Reviewer: Can only link to public classes
```suggestion
- Add ``save_as_surface`` method to allow saving GIFTI files from :class:`~mne.SourceEstimate` and related classes (:gh:`12309` by `Peter Molfese`_)
```
